### PR TITLE
Improve CQM documentation

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

File: dimod/constrained.py [hunk 1]
```diff
@@ -85,16 +89,114 @@ def vartype(self, v: Variable) -> Vartype:
...
+            & \sum_i a_i^{(d)} x_i + \sum_{i<j} b_{ij}^{(d)} x_i x_j + c^{(d)} = 0,
+            \quad d=1, \dots, C_{\rm eq.},
+        \end{align}
+
+    where :math:`\{ x_i\}_{i=1, \dots, N}` can be binary, integer or discrete
+    variables, :math:`a_{i}, b_{ij}, c` are real values and
+    :math:`C_{\rm ineq.}, C_{\rm eq,}` are the number of inequality and
+    equality constraints respectively.
+
+    The objective and constraints as encoded as either :class:`.QuadraticModel`
```
Comment (JoelPasvolsky) [inline dimod/constrained.py:112]: "The objective and constraints as encoded" --> are

File: dimod/constrained.py [hunk 2]
```diff
@@ -85,16 +89,114 @@ def vartype(self, v: Variable) -> Vartype:
...
+    or :class:`.BinaryQuadraticModel` depending on the variable types uses.
```
Comment (JoelPasvolsky) [inline dimod/constrained.py:113]: uses --> typo

File: dimod/constrained.py [hunk 3]
```diff
@@ -85,16 +89,114 @@ def vartype(self, v: Variable) -> Vartype:
...
+        generate bin packing problems. We follow the same naming conventions
+        in this example.
+
+        Let's start with four object weights and assume that each bin has a
+        capacity of 1.
+
+        >>> weights = [.9, .7, .2, .1]
+        >>> capacity = 1
+
+        Let :math:`y_i` indicate that we used bin :math:`j`. We know that we
```
Comment (JoelPasvolsky) [inline dimod/constrained.py:131]: y_i --> j ? y_j perhaps?

File: docs/reference/constrained.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   ConstrainedQuadraticModel.variables
+
+Methods
+-------
+
+.. autosummary::
+   :toctree: generated/
+
+   ConstrainedQuadraticModel.add_constraint
+   ConstrainedQuadraticModel.add_constraint_from_model
```
Comment (JoelPasvolsky) [inline docs/reference/constrained.rst:33]: from_m comes after from_iterable

File: docs/reference/index.rst [hunk 5]
```diff
@@ -9,6 +9,7 @@ Reference Documentation
    :maxdepth: 2
 
    bqm
+   constrained
```
Comment (JoelPasvolsky) [inline docs/reference/index.rst:12]: I'd put this first actually

**Review by JoelPasvolsky — APPROVED** [review]
LGTM. 
I'll probably rephrase some of the example wording later

File: docs/reference/index.rst (hunk 5 above)
Comment (arcondello) [reply docs/reference/index.rst:12]: I'll let you rearrange this stuff.

**Pull request merged**